src/widgets/Tabs/PatientEdit.js

```diff
@@ -106,23 +117,32 @@ const PatientEditComponent = ({
 const mapDispatchToProps = dispatch => {
   const onSubmitSurvey = formData => dispatch(NameNoteActions.saveNewSurvey(formData));
   const onCancelPrescription = () => dispatch(VaccinePrescriptionActions.cancel());
+  const updateForm = (data, errors) => dispatch(NameNoteActions.updateForm(data, errors));
   const updatePatientDetails = detailsEntered =>
     batch(() => {
       dispatch(NameActions.updatePatient(detailsEntered));
       dispatch(WizardActions.nextTab());
     });
 
-  return { onCancelPrescription, onSubmitSurvey, updatePatientDetails };
+  return { onCancelPrescription, onSubmitSurvey, updatePatientDetails, updateForm };
 };
 
 const mapStateToProps = state => {
   const currentPatient = selectEditingName(state);
   const completedForm = selectCompletedForm(state);
-  const canSaveForm = selectCanSaveForm(state);
+  const canSaveForm = selectCanSaveForm(state) && selectNameNoteIsValid(state);
```

Comment:
there's still an issue with the validation and seeding in the `Name` form - first name, last name and DOB are required; even when populated with the existing data, the validation isn't firing, so the button isn't enabled.